js/style/style.js

```diff
@@ -68,10 +76,32 @@ function Style(stylesheet, animationLoop) {
         this.fire('load');
     }.bind(this);
 
-    if (typeof stylesheet === 'string') {
-        ajax.getJSON(normalizeURL(stylesheet), loaded);
+    var sourceTypesLoaded = function(err) {
```

Comment:
Can you explain this part of the code? It looks like its loading support for core source types asynchronously? That seems odd to me, both that it should be async, and that something like `addSourceType` should be the responsibility of `Style` and attached as a non-underscored method.